lib/PersonalityProvider.jsm

```diff
@@ -71,7 +147,18 @@ this.PersonalityProvider = class PersonalityProvider {
 
   async getFromRemoteSettings(name) {
     const result = await RemoteSettings(name).get();
-    return result;
+    return Promise.all(result.map(async record => {
+      const {attachment: {filename, size}} = record;
+      // Figure out hash + md5sum check.
+      // const {attachment: {filename, hash, size}} = record;
+      await OS.File.makeDir(OS.Path.join(OS.Constants.Path.localProfileDir, PERSONALITY_PROVIDER_DIR_NAME));
+      const filepath = OS.Path.join(OS.Constants.Path.localProfileDir, PERSONALITY_PROVIDER_DIR_NAME, filename);
+      // Figure out hash + md5sum check.
+      if (!await OS.File.exists(filepath) || await OS.File.stat(filepath).size !== size) {
+        await this.downloadAttachment(record);
+      }
+      return {...await this.getAttachment(record), recordKey: record.key};
```

Comment:
Maybe here we'd want another safety check (redundant I know), because `this.getAttachment()` CAN return `null`

```suggestion
      const attachment = await this.getAttachment(record);
      if (!attachment) {
          throw new Error("Model data is missing and could not be downloaded");
      }
      return {...attachment, recordKey: record.key};
```

Reply:
Good catch, instead of a safety check, how about just making it so `getAttachment` returns `{}` instead of `null` in the previous `null` cases?

Reply:
This code seems to have been changed since you commented, so I think while what you said still applies about the null return being an issue, it's an issue in slightly different code now.

Reply:
The suggested changes feature threw me off for a sec, I thought I was looking at my diff.